packages/gatsby/src/bootstrap/remove-stale-jobs.ts

```diff
@@ -23,13 +23,18 @@ export const removeStaleJobs = (
 
   // If any of our pending jobs do not have an existing inputPath or the inputPath changed
   // we remove it from the queue as they would fail anyway
-  state.jobsV2.incomplete.forEach(({ job, plugin }: IGatsbyJobV2): void => {
-    if (isJobStale(job)) {
-      actions.push(internalActions.removeStaleJob(job.contentDigest))
-    } else {
-      actions.push(publicActions.createJobV2(job, plugin))
+  state.jobsV2.incomplete.forEach(
+    ({
+      job = { contentDigest: ``, inputPaths: [] },
```

Comment:
It's not necessary to have a default value here. There will never be undefined jobs, because the map uses job.contentDigest as the key. If the job was undefined then there would be no key available to insert it. If the job is not found in the Map then there's no need to remove or create it, so you can just return early